website/blog/2023-05-04-generating-dynamic-docs.md

```diff
@@ -0,0 +1,368 @@
+tier_threshold_amount
+user_id
+```
+
+Breaking down this command:
+- `grep ' \- name:' models/core/activity_based_interest/_activity_based_interest_docs.yml` searches for the pattern ` - name:` in the file `_activity_based_interest_docs.yml` located in the directory `models/core/activity_based_interest/`.
+- `cut -c 15-` removes the first 14 characters of each line from the output.
+- `sort -u` sorts the output in alphabetical order and removes any duplicate lines.
+
+### Format to align with Jinja docs block
+
+Copy-paste the above output into your `.md` file, so it looks like the following: 
+
+```
+$ cat models/core/activity_based_interest/_activity_based_interest_docs.md
+end_date
+event_day
+event_time
+id
+is_active
+last_updated_date
+start_date
+tier_interest_percentage
+tier_threshold_amount
+user_id
+```
+Now, open your code editor, and replace `(.*)` with `{% docs column__activity_based_interest__$1 %}\n\n{% enddocs %}\n`, which will result in the following in your markdown file:
+
+<Lightbox src="/img/blog/2023-05-04-generating-dynamic-docs/3.png" title="Replace content in your markdown file" />
+
+Now you can add documentation to each of your columns.
+
+## Update `.yml` file to source documentation from the `.md` file
+
+We can programmatically identify all columns, and have them point towards the newly-created documentation. In your code editor, replace `\s{6}- name: (.*)\n        description: ""` with `      - name: $1\n        description: "{{ doc('column__activity_based_interest__$1') }}`:
+
+<Lightbox src="/img/blog/2023-05-04-generating-dynamic-docs/4.png" title="Replace descriptions with dynamic doc blocks" />
+
+⚠️ Some of your columns may have already been documented in other domains. In our example, we want to make the following replacements:
+- `{{ doc('column__activity_based_interest__user_id') }}` → `{{ doc("column_user_id") }}`
+- `{{ doc('column__activity_based_interest__event_day') }}` → `{{ doc("column_event_day") }}`
+
+## Check that everything works
+Now we run `dbt docs generate`. If there are syntax errors, this will be found out at this stage. If successful, we can run `dbt docs serve` to perform a smoke test and ensure everything looks right:
+
+<Lightbox src="/img/blog/2023-05-04-generating-dynamic-docs/5.jpg" title="dbt Docs UI showing successful documentation leveraging docs blocks" />
+
+## Additional considerations
+
+- Q: What about when they are slight deviations in column documentation between models?
+    - A: I’ve been using dynamic documentation to contain the "essence" of the documentation, and then appending static documentation, like so:
+
+    ```yaml
+           - name: user_id
+             description: "{{ doc('dynamic_docs') }}, additional static info" 
+    ```
+- Q: Should I use this approach on modifications to an existing domain? 
+    - A: When adding additional models, or additional columns to an existing domain, I would suggest adding new documentation manually.
```

Comment:
```suggestion
- Q: Should I use this approach on modifications to an existing folder? 
    - A: When adding additional models to a folder, or additional columns to an existing model, I would suggest adding new documentation and docs blocks manually rather than programmatically.
```